mart/attack/adversary.py

```diff
@@ -315,11 +328,11 @@ def forward(
         # Generate a perturbation only if we have a model. This will update
         # the parameters of self.perturber.
         if model is not None:
-            super().forward(input=input, target=target, model=model, **kwargs)
+            self.attacker.fit(input=input, target=target, model=model, **kwargs)
 
         # Get perturbation and apply threat model
         # The mask projector in perturber may require information from target.
-        perturbation = self.perturber(input, target)
+        perturbation = self.attacker(input=input, target=target)
         output = self.composer(perturbation, input=input, target=target)
```

Comment:
This feels like we should integrate `Composer` into `Attacker` and have `Attacker` returned adversarial inputs?

Reply:
@mzweilin: any thoughts?